urwid/tests/test_signals.py

```diff
@@ -25,6 +26,7 @@ def test_connect(self):
         edit.set_edit_text('another text')
         handler.assert_not_called()
 
+    @unittest.skipIf(sys.implementation.name == "pypy", "WeakRef works differently on PyPy")
```

Comment:
this test is expecially unstable on local PyPy run.
Also manual experiments shown, that everything garbage collector related work can work differently on PyPy